### ISSUE TITLE
Accessibility: unable to access radio buttons or checkboxes whilst using a screen reader

### ISSUE THREAD

**Issue by chamandeep** [davidstutz/bootstrap-multiselect#407]
The dropdown menu is not accessible to screen readers by default. When radio buttons or checkboxes are displayed, they are contained within an <a> tag. This means that when screen readers come across this element, they see it as a hyperlink and therefore are unable to tick the checkbox or radio button.
I have investigated into several solutions to this and have found that the <a> tag is required in the plugin in order for keyboard accessibility to be maintained.
Expected result is to be able to use this with the keyboard and screen readers independantly. The screen reader must be able to interact with the checkbox or radio button directly. I have also attempted to set the role of 'radio' or 'checkbox' based on the multiple attribute, but this shows the user 2 duplicate elements.

**Comment by rps1007** [comment]
@chamandeep : Is there a workaround to resolve this? You have mentioned a tag is needed . Can you pls let me know what tag is needed ?

@davidstutz : Is this enhancement already available in the latest version ?

**Comment by hrvojegolcic** [comment]
If anyone have an idea about the task I'd be happy to contribute

**Comment by hanatharesh2712** [comment]
@davidstutz any update on this issue? Thanks!

**Comment by hanatharesh2712** [comment]
@chamandeep the issue will be resolved by adding `role="combobox"` in button element here:  [https://github.com/davidstutz/bootstrap-multiselect/blob/63a9191d04f5c2eebb7e64e102a8295c92feb7d9/dist/js/bootstrap-multiselect.js#L442](url)

**Comment by Aryan3212** [comment]
Hi everyone, seems like I hacked the problem. The dropdown is created from a skeleton of select and option tags and then rendered onto the screen as button>span>input+span. I randomly found that if I used a an aria-labelled="ID" tag with the button that points to the ID of the input tag **and** the input tag has a random aria-label, let's assume aria-label="1". The NVDA screen reader just seems to work, even though it's still a bit buggy.

`function accessibleDropdown() {
                const dropdownChildrenList= document.querySelector('span.multiselect-native-select div.btn-group div.multiselect-container').children;
                dropdownChildrenList.map(makeAccessible);
            }
            
            function makeAccessible(index, element) {
               
                const inputElem = element.querySelector('input');
                
                element.setAttribute('aria-labelledby',inputElem.id);
                inputElem.setAttribute('aria-label','1');
               
            }
            `

**Comment by hrvojegolcic** [comment]
@Aryan3212 did you actually test this with all screen readers? I'd strongly advise not to use such "hacks" as this can make the whole component inaccessible for some users. The following pattern is what works correctly atm: https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html

**Comment by Aryan3212** [comment]
@hrvojegolcic Hi, I tested it with NVDA, but yes it would be even worse for accessibility if it wasn't tested across the board. I only intended to steer the conversation to some direction that would lead to a solution. A hack isn't really a solution. I didn't know what the W3 standard talked about, thanks a lot for sharing 😃
